planner/core/logical_plan_builder.go

```diff
@@ -2775,11 +2782,11 @@ func (b *PlanBuilder) resolveHavingAndOrderBy(ctx context.Context, sel *ast.Sele
 	}
 	havingAggMapper := extractor.aggMapper
 	extractor.aggMapper = make(map[*ast.AggregateFuncExpr]int)
-	extractor.inExpr = false
 	// Extract agg funcs from order by clause.
 	if sel.OrderBy != nil {
 		extractor.curClause = orderByClause
 		for _, item := range sel.OrderBy.Items {
+			extractor.inExpr = false
```

Comment:
this is for two order by item says:
[year+2, profit]
it will identify profit column with flag 'IN EXPR', just fix it here tooo